src/main/java/jenkins/branch/Branch.java

```diff
@@ -42,6 +42,7 @@
 
 /**
  * A source code branch.
+ * Not to be subclassed outside this plugin.
```

Comment:
I was thinking more something like `assert getClass().getPackage() == Branch.getClass().getPackage()` or otherwise asserting that the subclass is within the same jar as Branch.class